.dprint.jsonc

```diff
@@ -0,0 +1,52 @@
+{
+    "indentWidth": 4,
+    "lineWidth": 1000,
+    "newLineKind": "auto",
+    "useTabs": false,
+    "typescript": {
+        "semiColons": "always",
+        "quoteStyle": "preferDouble",
+        "useBraces": "whenNotSingleLine",
+        "bracePosition": "sameLineUnlessHanging",
+        "singleBodyPosition": "sameLine",
+        "nextControlFlowPosition": "nextLine", // Stroustrup style braces.
+        "trailingCommas": "onlyMultiLine",
+        "preferHanging": false,
+        "operatorPosition": "maintain",
+
+        "arrowFunction.useParentheses": "preferNone",
+        "conditionalExpression.linePerExpression": false, // Keep our "match/case"-ish conditionals.
+        "functionExpression.spaceAfterFunctionKeyword": true,
+        "importDeclaration.forceMultiLine": true,
+        "constructorType.spaceAfterNewKeyword": true,
+        "constructSignature.spaceAfterNewKeyword": true,
+
+        // Let eslint-plugin-simple-import-sort handle this.
+        "module.sortImportDeclarations": "maintain",
+        "module.sortExportDeclarations": "maintain",
+        "exportDeclaration.sortNamedExports": "maintain",
+        "importDeclaration.sortNamedImports": "maintain"
+    },
+    "prettier": {
+        "yml.tabWidth": 2,
+        "yaml.tabWidth": 2,
+        "yml.singleQuote": true,
+        "yaml.singleQuote": true
+    },
+    "includes": ["**/*.{ts,tsx,mts,cts,js,jsx,cjs,mjs,yml,yaml}"],
+    "excludes": [
+        "**/node_modules",
+        "**/*-lock.json",
+        "coverage/**",
+        "lib/**",
+        "built/**",
+        "tests/**",
+        "internal/**",
+        "**/*.generated.*",
+        "scripts/*.d.*"
+    ],
+    "plugins": [
+        "https://plugins.dprint.dev/typescript-0.85.1.wasm",
+        "https://plugins.dprint.dev/prettier-0.26.1.json@fdbe31f6aecd24f9d6b924214710a6766050d03146163b4e241e6056b2462f2e"

```

Comment:
This is only used for yaml formatting, right? One thing I'd recommend is to remove this for the first pass because at the moment this supports less platforms since it will download a separate binary (with Prettier snapshotted in v8 in it). I can write a yaml formatter in Rust because I don't think it would be too much work, then compile that to Wasm for use here (which is much more portable). I've wanted to do that anyway.

Reply:
Yes, that's correct, this is for yaml only.

Reply:
I'm not sure this is going to run anywhere besides amd64 or macos arm64, so if those are supported then I'd rather do this than not; I've so far been plugging all of my yml into prettier.io and copying it back...

Reply:
Sounds good! Yeah I doubt you’ll have trouble, but if you do then it’s easy to temporarily remove it.